resource/graph_node.go

```diff
@@ -299,7 +302,7 @@ func (w *GraphNode) Config() Config {
 func (w *GraphNode) NeedsReconfigure() bool {
 	w.mu.RLock()
 	defer w.mu.RUnlock()
-	return w.state == NodeStateConfiguring
+	return w.state == NodeStateConfiguring || w.state == NodeStateUnhealthy
```

Comment:
a resource only enters unhealthy states during (re)configuration

Reply:
I think that would be valuable information as a code comment